test/test_python_dispatch.py

```diff
@@ -0,0 +1,139 @@
+import torch
+from torch.testing._internal.common_utils import TestCase, run_tests
+from torch._C import _disabled_torch_function_impl
+
+from typing import Iterator, List, Dict
+import logging
+import contextlib
+
+# TODO: TensorBase should work
+class LoggingTensor(torch.Tensor):
+    elem: torch.Tensor
+
+    @staticmethod
+    def __new__(cls, elem, *args, **kwargs):
+        r = torch.Tensor._make_subclass(cls, elem.to('meta'), elem.requires_grad)
+        r.elem = elem
+        return r
+
+    def __repr__(self):
+        return f"LoggingTensor({self.elem})"
+
+    def __str__(self):
+        return f"LoggingTensor({self.elem})"
+
+    def __format__(self, format_spec):
+        return f"LoggingTensor({self.elem})"
+
+    __torch_function__ = _disabled_torch_function_impl
+
+    @classmethod
+    def __torch_dispatch__(cls, func, types, args=(), kwargs=None):
```

Comment:
Yeah, if we decide to never pass kwargs (I'm still not 100% clear what we should do here), we should modify the signature here. It's a little annoying because I have to add more conditional swiss cheese.